packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,287 @@
+import ExpoSQLiteDatabase from '../src/ExpoSQLiteAdapter/ExpoSQLiteDatabase';
+import { openDatabase } from 'expo-sqlite';
+import { ParameterizedStatement } from '../src/common/types';
+import { HelperStack } from 'typedoc/dist/lib/output/utils/resources/helpers';
+
+// mock out the underlying implementation and only keep openDatabase().
+jest.mock('expo-sqlite', () => {
+	return {
+		__esModule: true,
+		openDatabase: jest.fn(() => mockDatabase),
+	};
+});
+
+// mock out the underlying implementation
+jest.mock('expo-file-system', () => 'File System');
+
+const mockDatabase = {
+	readTransaction: jest.fn(),
+	transaction: jest.fn(),
+};
+
+const mockTransaction = {
+	executeSql: jest.fn(),
+};
+
+const mockResultSet = jest.fn();
+
+const MOCK_SQL_ERROR = {
+	code: 400,
+	message: 'Error performing the transaction',
+};
+
+describe('ExpoSQLiteAdapter', () => {
+	const ExpoSQLiteAdapterInstance = new ExpoSQLiteDatabase();
+	ExpoSQLiteAdapterInstance.init();
+
+	beforeEach(() => {
+		mockDatabase.readTransaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
+		mockDatabase.transaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
+
+		// mockImplementation of executeSql first two calls should call successCallback, followed by errorCallback
+		// and remaining all execution wil call successCallback
+		// intended to check behavior if promises are resolved and rejected correctly
+		mockTransaction.executeSql
```

Comment:
Probably best to just mock the implementation at the test level instead of trying to make every test conform to a sequence of events.

Reply:
Cool. I have made those changes.